src/commonMain/kotlin/kotools/types/collection/NotEmptyList.kt

```diff
@@ -54,6 +39,30 @@ public fun <E> Collection<E>.toNotEmptyList(): Result<NotEmptyList<E>> =
         }
         ?: Result.failure(EmptyCollectionException)
 
+/** Representation of lists that contain at least one element of type [E]. */
+@JvmInline
+@Serializable(NotEmptyListSerializer::class)
+@SinceKotoolsTypes("4.0")
+public value class NotEmptyList<out E> private constructor(
+    private val elements: List<E>
+) : NotEmptyCollection<E> {
+    override val head: E get() = elements.first()
+    override val tail: NotEmptyList<E>?
```

Comment:
We should improve the documentation of this operation by adding usage examples.

Reply:
Done by improving the documentation of the `NotEmptyCollection.tail` property (see commit 56c1c09210975e0c628557cd7c33f7816104b3d9).